### ISSUE TITLE
Following mu4e-org link before mu4e has been used causes confusing errors

### ISSUE THREAD

**Issue by bcc32** [djcb/mu#2792]
## Expected or desired behavior

Opening an org-mode link like the following:

    [[mu4e:query:maildir:"/INBOX"]]

Should open a `*mu4e-headers*` buffer with the desired query, even if mu4e
hasn't been used yet in the current Emacs session.

## Actual behavior

The `*mu4e-headers*` buffer is empty and the following error is issued (lines
truncated because they are very, very long):

```
Debugger entered--Lisp error: (void-function nil)
  nil(((:path "/home/aaron/mail/Newslette..." :size 122879 :
  funcall(nil ((:path "/home/aaron/mail/Newslette..." :size
  (cond ((plist-get sexp :headers) (funcall mu4e-headers-app
  (while sexp (mu4e-log 'from-server "%s" sexp) (cond ((plis
  (let ((inhibit-quit nil)) (while sexp (mu4e-log 'from-serv
  (condition-case nil (let ((inhibit-quit nil)) (while sexp
  (let ((sexp (mu4e--server-eat-sexp-from-buf))) (condition-
  mu4e--server-filter(#<process  *mu4e-server*> "\376b\377(:
```

In particular, `mu4e--server-filter` is attempting to funcall
`mu4e-headers-append-func`, but that value is still nil.

Also, it is inconvenient to remedy this problem because after it occurs, `M-x
mu4e` does not fix the issue; instead it errors:

```
mu4e-error: [mu4e] Root maildir unknown; did you start mu4e?
```

## Steps to reproduce

1. emacs -Q, M-x package-initialize, M-: (require 'mu4e-org)
2. Open an org-mode buffer and paste in this link

    [[mu4e:query:maildir:"/INBOX"]]

3. With point at the link, press C-c C-o.

## Versions of mu, mu4e/emacs, operating system etc.

mu4e-mu-version is 1.12.5.  Observed this issue on both NixOS and macOS.

GNU Emacs 29.4 (build 1, x86_64-pc-linux-gnu, X toolkit, cairo version 1.18.0, Xaw3d scroll bars)

**Comment by djcb** [comment]
Can't reproduce.

Created file `test.org` with some lines:
```
(add-to-list 'load-path "/home/djcb/Sources/mu/build/mu4e")
;; set mu4e-mu-binary / mu4e-mu-home if needed
(require 'mu4e-org)
[[mu4e:query:maildir:"/INBOX"]]
```
then `emacs -Q test.org`; evaluate first to lines, then C-o C-o and works just fine.

So I suspect some setup issue.

**Comment by bcc32** [comment]
Yes, I get the same error with your `test.org`.  Of course, I had to change the load path to:

```
(add-to-list 'load-path "~/.nix-profile/share/emacs/site-lisp/elpa/mu4e-1.12.5/")
```

but no other changes.

**Comment by djcb** [comment]
It's weird, at the top of  `mu4e-org-open` we have:
```
(defun mu4e-org-open (link)
  "Open the org LINK.
Open the mu4e message (for links starting with \"msgid:\") or run
the query (for links starting with \"query:\")."
  (require 'mu4e)
  (mu4e 'background)
...
```
and this initializes `mu4e-headers-append-func` etc. So I'm not sure why that's not working in your case.

Does Nix setup mu4e in some special way?

**Comment by djcb** [comment]
Ah, found it. You're on 1.12.5, but this was fixed in 1.12.7 (`18a9bd4a`). So please upgrade.

**Comment by bcc32** [comment]
Ah, thanks for looking into this, and sorry for the false alarm!
